cmd/snap-gdb-shim/snap-gdb-shim.c

```diff
@@ -29,6 +29,13 @@ int main(int argc, char **argv)
 			printf("-%s-\n", argv[i]);
 		}
 	}
+        // check if we run as SUDO and if so switch to a normal user
+        const char *sudo_uid_env = getenv("SUDO_UID");
+        if (sudo_uid_env != NULL) {
+           int sudo_uid = atoi(sudo_uid_env);
```

Comment:
Could you please use `strtol` instead of `atoi` and check for errors?
We have an example in the tree as quick reference:

```
        char *endptr = NULL;
        errno = 0;
        long env_max_wait = strtol(MAX_PROFILE_WAIT, &endptr, 10);
        if (errno != 0 || MAX_PROFILE_WAIT == endptr || *endptr != '\0'
            || env_max_wait <= 0) {
            die("SNAP_CONFINE_MAX_PROFILE_WAIT invalid");
        }
```

Reply:
`strtoul` even?